### PR TITLE
busybox: 1.27.2 -> 1.28.0

### PR DESCRIPTION
**Pull request by adisbladis**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc maintainer @viric

### CODE REVIEW

**Comment by dtzWill** [comment]
LGTM.  bootstrap tools still build fine, which is usually the concern here.

File: pkgs/os-specific/linux/busybox/busybox-in-store.patch [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
+--- a/libbb/appletlib.c
++++ b/libbb/appletlib.c
+@@ -947,7 +947,7 @@ void FAST_FUNC run_applet_no_and_exit(int applet_no, const char *name, char **ar
  static NORETURN void run_applet_and_exit(const char *name, char **argv)
  {
  #  if ENABLE_BUSYBOX
 -	if (is_prefixed_with(name, "busybox"))
-+	if (strstr(name, "busybox") != 0)
- 		exit(busybox_main(argv));
++       if (strstr(name, "busybox") != 0)
```
Comment (andir) [inline pkgs/os-specific/linux/busybox/busybox-in-store.patch]: probably a nitpick but the indention is a bit off here

File: pkgs/os-specific/linux/busybox/busybox-in-store.patch [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
  #  if NUM_APPLETS > 0
-@@ -981,7 +981,7 @@ int main(int argc UNUSED_PARAM, char **argv)
- 
+@@ -1045,7 +1045,7 @@ int main(int argc UNUSED_PARAM, char **argv)
+
  	lbb_prepare("busybox" IF_FEATURE_INDIVIDUAL(, argv));
  # if !ENABLE_BUSYBOX
 -	if (argv[1] && is_prefixed_with(bb_basename(argv[0]), "busybox"))
-+	if (argv[1] && strstr(bb_basename(argv[0]), "busybox") != 0)
++       if (argv[1] && strstr(bb_basename(argv[0]), "busybox") != 0)
```
Comment (andir) [inline pkgs/os-specific/linux/busybox/busybox-in-store.patch]: probably a nitpick but the indention is a bit off here

File: pkgs/os-specific/linux/busybox/default.nix [hunk 3]
```diff
@@ -27,35 +27,20 @@ let
 in
 
 stdenv.mkDerivation rec {
-  name = "busybox-1.27.2";
+  name = "busybox-1.28.0";
```
Comment (andir) [inline pkgs/os-specific/linux/busybox/default.nix]: All of the security patches that you removed are in this release. So nothing to worry here :+1:

**Review by andir — COMMENTED** [review]
LGTM. Depending on your motivation you can fix the indention issues ;-)

**Review by NeQuissimus — APPROVED** [review]

**Comment by andir** [comment]
@GrahamcOfBorg eval

**Comment by adisbladis** [comment]
This was pushed in 65774cb22e0b40a3ac7bc97adde3767f1d7df20b

**Pull request closed without merging**